### PR TITLE
[4.0] Revert helper

### PR DESCRIPTION
**Pull request by bembelimen**
Pull Request for Issue #21518 .

### Summary of Changes
Removed unused helper methods and update count methods

@wilsonge

### CODE REVIEW

**Comment by wilsonge** [comment]
let's leave archive here please

**Comment by bembelimen** [comment]
At the current status it makes no sense to habe it here, this will be readded with the refactoring, I think.

**Comment by infograf768** [comment]
Any change introduced by Workflow should be a conditional depending on the setting "Enable Workflows" for articles.
I.e. Nothing that existed before should be deleted/modified to let users choose to use workflow or not.

This means that the priority is to recover what has been deleted by workflow.

**Comment by brianteeman** [comment]
There is NO option to disable/enable workflows. You have no choice - this workflows pr completely replaces the current working system. 

That is why I am so insistent that the code is correct. The only reason that the current/previous states still exist in the com-content table is for b/c with extensions eg content-modules. Although that is not true either as it will only write published/unpublished/trashed into that table and will not write archived.

The option that you are thinking of works the same as the one for custom fields. ie it hides/shows the menu admin links. The logic being that once you have workflows setup then you don't need to see it again. Yes the language string for both needs to be improved.

If people spent the time to truly test this workflows stuff in a real world scenario they would see why it is not ready at all and will require fundamental changes before it can even be considered. The fact that it ignores the ACL is the most critical to me.

File: administrator/components/com_content/Helper/ContentHelper.php [hunk 1]
```diff
@@ -200,6 +201,9 @@ public static function countTagItems(&$items, $extension)
 			$item->count_trashed     = 0;
 			$item->count_unpublished = 0;
 			$item->count_published   = 0;
+			$item->count_archived   = 0;
+
+
```
Comment (Quy) [inline administrator/components/com_content/Helper/ContentHelper.php]: Remove blank line.

File: administrator/components/com_content/Helper/ContentHelper.php [hunk 2]
```diff
@@ -115,6 +115,7 @@ public static function countItems(&$items)
 			$item->count_trashed     = 0;
 			$item->count_unpublished = 0;
 			$item->count_published   = 0;
+			$item->count_archived   = 0;
```
Comment (Quy) [inline administrator/components/com_content/Helper/ContentHelper.php]: Align `=`

**Comment by bembelimen** [comment]
@wilsonge I readded the initial archive variables and added the condition constants.

For now this is the current working status for merging. Archived will be improved (again), when @HLeithner finished the break up of the conditions (which is the next step for this PR)

**Pull request merged**